mxcubecore/HardwareObjects/ICATLIMS.py

```diff
@@ -410,11 +411,13 @@ def _store_data_collection_group(self, group_data):
         pass
 
     @property
-    def only_staff_session_selection(self):
-        return bool(self.get_property("only_staff_session_selection", default=False))
+    def only_staff_session_selection(self) -> bool:
+        return bool(
+            self.get_property("only_staff_session_selection", default_value=False)
+        )
 
     def store_robot_action(self, proposal_id: str):
-        raise Exception("Not implemented")
+        raise NotImplementedError("Not implemented")
```

Comment:
If I understood correctly, this could be simplified to this:

```suggestion
        raise NotImplementedError
```

I do not think the message is necessary, because it does not add any new information since the exception type is sufficiently explicit already. And then the braces can be omitted, as far as I know raising an exception class or an exception instance has the same effect.

Just an FYI.

Reply:
Agree.